### PR TITLE
Refactor Job.FileStore into fileStore.py (resolves #1028)

### LINKED ISSUE #1028: Extract Job.FileStore into separate module
FileStore was your average guy, living in a luxurious space with a Job(.py), and a lot of responsibility. Everything changed when the new kid on the block (CachedFileStore) decided to cause trouble in his neighborhood. The new kid was smart... he observed FS and imbibed many of his properties. What the young CFS didn't realize is that these actions would bind the two forever.... or so it seemed.

Now, with the neighborhood getting too crowded, the duo decided to move to the countryside to their new estate at fileStore.py.

To be continued

TL;DR - refactoring FileStore and CachedFileStore into fileStore.py because Job is getting too large.

### PR DESCRIPTION
**Pull request by arkal**
resolves #1028

### CODE REVIEW

**Comment by arkal** [comment]
Jenkins, test this please.

**Comment by arkal** [comment]
@hannes-ucsc this is ready!

File: src/toil/fileStore.py [hunk 1]
```diff
@@ -0,0 +1,1372 @@
...
+
+from toil.common import cacheDirName
+from toil.lib.bioio import makePublicDir
+
+logger = logging.getLogger( __name__ )
+
+
+class FileStore(object):
+    '''
+    Class used to manage temporary files, read and write files from the job store\
```
Comment (hannes-ucsc) [inline src/toil/fileStore.py]: Remove \

Comment (hannes-ucsc) [inline src/toil/fileStore.py]: Never mind. I'd rather copy the mistakes in this PR and fix them in a later one.

File: src/toil/test/src/jobCacheTest.py [hunk 2]
```diff
@@ -37,6 +38,7 @@
 testingIsAutomatic = True
 
 
+# noinspection PyClassHasNoInit
```
Comment (hannes-ucsc) [inline src/toil/test/src/jobCacheTest.py]: Can you remove this and disable the inspection in your IDE instead? I don't think this is a valid inspection in the first place. I can't see why a class not having a constructor could be a problem in Python.

**Comment by hannes-ucsc** [comment]
Just that tiny inspection thing. And can you confirm for all eternity that you didn't touch the actual FileStore code in this PR?

File: src/toil/test/src/jobCacheTest.py (hunk 2 above)
Comment (arkal) [inline src/toil/test/src/jobCacheTest.py]: haha

**Comment by arkal** [comment]
i updated 2 docstrings.... is that a problem?

**Comment by hannes-ucsc** [comment]
No

**Comment by arkal** [comment]
@hannes-ucsc this is ready!

**Pull request merged**